### PR TITLE
refactor text graphic

### PR DESCRIPTION
**Pull request by clewis7**
#305

Refactored the text graphic to use properties and some more features of `pygfx TextGeometry`

### CODE REVIEW

**Comment by clewis7** [comment]
Screen Shot 2023-10-25 at 3 20 09 PM

File: fastplotlib/graphics/text.py [hunk 1]
```diff
@@ -43,11 +44,16 @@ def __init__(
...
 
         """
+        super(TextGraphic, self).__init__(*args, **kwargs)
 
-        super(TextGraphic, self).__init__(name=name)
+        self._text = text
 
         world_object = pygfx.Text(
-            pygfx.TextGeometry(text=str(text), font_size=size, screen_space=False),
+            pygfx.TextGeometry(text=str(text),
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:52]: can you use `black` for now

File: fastplotlib/graphics/text.py [hunk 2]
```diff
@@ -43,11 +44,16 @@ def __init__(
...
+                               font_size=size,
+                               screen_space=True,
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:54]: make a kwarg for this?

File: fastplotlib/graphics/text.py [hunk 3]
```diff
@@ -43,11 +44,16 @@ def __init__(
...
+                               anchor="middle-center"
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:55]: make a kwarg for this

File: fastplotlib/graphics/text.py [hunk 4]
```diff
@@ -43,11 +44,16 @@ def __init__(
...
+                               ),
             pygfx.TextMaterial(
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:57]: any other useful kwargs to have for material?

File: fastplotlib/graphics/text.py [hunk 5]
```diff
@@ -59,22 +65,78 @@ def __init__(
...
-    def update_face_color(self, color: Union[str, np.ndarray]):
+    @property
+    def face_color(self):
+        """Returns the face color of this graphic."""
+        return self.world_object.material.color
+
+    @face_color.setter
+    def face_color(self, color: Union[str, np.ndarray]):
+        """Set the face color of this graphic."""
+        if not (isinstance(color, str) or isinstance(color, np.ndarray)):
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:103]: I think the text material color should be able to take anything in the form of `pygfx.Color`, so just convert the input color to `pygfx.Color` and use that?

File: fastplotlib/graphics/text.py [hunk 6]
```diff
@@ -59,22 +65,78 @@ def __init__(
...
         self.world_object.material.outline_thickness = size
 
-    def update_outline_color(self, color: Union[str, np.ndarray]):
+    @property
+    def outline_color(self):
+        """Returns the outline color of this graphic."""
+        return self.world_object.material.outline_color
+
+    @outline_color.setter
+    def outline_color(self, color: Union[str, np.ndarray]):
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:127]: I think the text material color should be able to take anything in the form of pygfx.Color, so just convert the input color to pygfx.Color and use that?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: fastplotlib/graphics/text.py (hunk 4 above)
Comment (clewis7) [reply fastplotlib/graphics/text.py:57]: no, I don't think so...not anything that should be exposed to a user

**Comment by kushalkolar** [comment]
can you run black on only the textgraphic, else this will cause merge conflicts with every other WIP PR. 

we should look into ruff longterm

File: docs/source/api/graphics/TextGraphic.rst
Comment (kushalkolar) [inline docs/source/api/graphics/TextGraphic.rst:1]: the api docs need to be auto-generated by the `generate_api.py` script

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Comment by kushalkolar** [comment]
seems like the wrong commit got reverted, this is still changing the rest of the codebase

File: fastplotlib/graphics/text.py [hunk 7]
```diff
@@ -59,22 +77,80 @@ def __init__(
...
+        """Set the outline color of this graphic"""
+        if not (isinstance(color, str) or isinstance(color, np.ndarray)):
+            raise ValueError("Outline color must be of type str or np.ndarray")
+
         self.world_object.material.outline_color = color
 
-    def update_position(self, pos: Tuple[int, int, int]):
-        self.world_object.position.set(*pos)
+    @property
+    def text_position(self):
```
Comment (kushalkolar) [inline fastplotlib/graphics/text.py:149]: These aren't necessary, it inherits from Graphic which has position properties

**Review by kushalkolar — CHANGES_REQUESTED** [review]
Also regenerate the add graphics mixin

**Review by kushalkolar — APPROVED** [review]

**Pull request merged**